File: EERPROMWriter/writer/runner/matrix.py

matrix = {
"HLT": 0,
"LDA": 1,
"&LDA": 2,
"STO": 3,
"ADD": 4,
"SUB": 5,
"BRA": 6,
"BRZ": 7,
"BRC": 8,
"BRP": 9,
"&ADD": 10,
"&SUB": 11,
"CFR": 12,
"STF": 13,
"NOP": 14,
"OUT": 15,
"JMP": 16,
"INC": 17,
"DEC": 18,
"JPF": 19,
"INP": 20
}
def code(command):
    return matrix.get(command)
    

File: Emulator/src/Code/Matrix.java
package Code;

import java.util.HashMap;

public class Matrix {
    
    public static HashMap<String, Integer> matrix = new HashMap<String, Integer>(){{
    put("HLT", 0);
    put("LDA", 1);
    put("&LDA", 2);
    put("STO", 3);
    put("ADD", 4);
    put("SUB", 5);
    put("BRA", 6);
    put("BRZ", 7);
    put("BRC", 8);
    put("BRP", 9);
    put("&ADD", 10);
    put("&SUB", 11);
    put("CFR", 12);
    put("STF", 13);
    put("NOP", 14);
    put("OUT", 15);
    put("JMP", 16);
    put("INC", 17);
    put("DEC", 18);
    put("JPF", 19);
    }};

    public static int get(String key){
        return (matrix.containsKey(key)) ? matrix.get(key) : -1;  
    }
        
}
